tls/s2n_config.h

```diff
@@ -40,6 +45,16 @@ struct s2n_config {
     s2n_client_hello_fn *client_hello_cb;
     void *client_hello_cb_ctx;
 
+    uint64_t session_state_lifetime_in_nanos;
+
+    uint8_t use_tickets;
+    struct s2n_ticket_key ticket_keys[S2N_MAX_TICKET_KEYS];
```

Comment:
why make this a static array, especially if someone doesn't want to use tickets.

1Mb is a lot of memory, espicially on embedded platforms.

consider using the dynamic array that was added recently.

Reply:
Are you talking about using dynamic array for ticket_keys or ticket_key_hashes or both?

ticket_key_hashes is currently set to 1MB by default ( S2N_MAX_TICKET_KEY_HASHES 50000)
The maximum number of ticket_keys is just 48.

Reply:
both - or why not just use the map?

either way, a statically defined 1Mb array on the config object is going to upset a lot of people ;)